### ISSUE TITLE
False positive "[TypedDict instance] is not callable"

### ISSUE THREAD

**Issue by l0b0** [PyCQA/pylint#4715]
### Steps to reproduce

Given a file `a.py`:
```python
from typing import TypedDict

Link = TypedDict("Link", {"href": str})
Link(href="foo")
```

### Current behavior

Result of `pylint a.py`:
```
************* Module a
a.py:4:0: E1102: Link is not callable (not-callable)

------------------------------------
Your code has been rated at -6.67/10
```

### Expected behavior

This should not be an error. `TypedDict` is a subclass of `dict`, so line 4 is equivalent to `dict(href="foo")`. The code also runs fine and passes `mypy --strict a.py`.

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.8.10 (default, Jul  7 2021, 14:29:06) 
[GCC 11.1.0]
```

**Comment by cdce8p** [comment]
Thanks for the bug report! I've opened PRs to address the issue.

You can work around it already though by using the "new" syntax instead.
```py
from typing import TypedDict

class Link(TypedDict):
    href: str

Link(href="foo")
```

https://docs.python.org/3/library/typing.html#typing.TypedDict

**Comment by l0b0** [comment]
You fixed it *and* taught me something new. Thank you so much!
